### PR TITLE
Port onnxruntime quantizer

### PR DESCRIPTION
**Pull request by take-cheeze**

### CODE REVIEW

File: runtime/chainerx_util.cc [hunk 1]
```diff
@@ -265,5 +269,20 @@ chainerx::Array GroupedConv(
     return chainerx::Conv(x, w, b, strides, pads);
 }
 
+// TODO(take-cheeze): Implement in ChainerX
+chainerx::Array SlowRound(const chainerx::Array& x) {
+    std::vector<double> result_data(x.GetTotalSize());
+    chainerx::Array double_x = x.AsType(chainerx::Dtype::kFloat64);
+    const double* x_ptr = reinterpret_cast<const double*>(double_x.raw_data());
```
Comment (shinh) [inline runtime/chainerx_util.cc]: `CHECK(IsNativeDevice(&x.device()))`?

**Comment by take-cheeze** [comment]
Have some issue in generated ONNX nodes.
Seems it's different from quantizer script like:

I need ONNX diff...

```shell
# convert with onnxruntime quantizer using script
~/dev/chainer-compiler$ ./utils/quantize.sh --no_per_channel ./data/mnist/model.onnx ./data/mnist/q_model.onnx
Quantize config: {'nbits': 8, 'quantization_mode': 0, 'static': False, 'asymmetric_input_types': False, 'input_quantization_params': None, 'output_quantization_params': None, 'per_channel': False}
Saving "./data/mnist/model.onnx" to "./data/mnist/q_model.onnx"

# result of onnxruntime quantized result:
~/dev/chainer-compiler$ ./build/tools/run_onnx --onnx ./data/mnist/q_model.onnx --test ./data/mnist
Initializing ChainerX...
Loading model...
Loading data...
Found 3 test cases
Constructing model...
Generate code...
Running for ./data/mnist/test_data_set_0
Verifying the result...
FAIL(value): Plus214_Output_0
Expected: array([[  975.67010498,  -618.72393799,  6574.56835938,   668.02893066,  -917.27093506, -1671.63586426, -1952.75988770,   -61.54987335,  -777.17663574, -1439.53161621]], shape=(1, 10), dtype=float32, device='native:0')
Actual: array([[  970.61535645,  -610.13140869,  6588.47851562,   625.06878662,  -911.78411865, -1684.60876465, -1983.74023438,   -29.74928093,  -788.45288086, -1440.35058594]], shape=(1, 10), dtype=float32, device='native:0')
Mismatch: 10 / 10 (100%)
Check `ok_cnt' == `test_case.outputs.size()' failed! in VerifyOutputs at ../tools/run_onnx.cc:499: (0 vs 1)
Aborted (core dumped)

# quantized running in chainer-compiler:
~/dev/chainer-compiler$ ./build/tools/run_onnx --quantize ./data/mnist/
Initializing ChainerX...
Loading model...
Loading data...
Found 3 test cases
Constructing model...
Generate code...
Running for ./data/mnist//test_data_set_0
Verifying the result...
FAIL(value): Plus214_Output_0
Expected: array([[  975.67010498,  -618.72393799,  6574.56835938,   668.02893066,  -917.27093506, -1671.63586426, -1952.75988770,   -61.54987335,  -777.17663574, -1439.53161621]], shape=(1, 10), dtype=float32, device='native:0')
Actual: array([[-1898.08447266, -1102.85778809,  1229.7224121 ,  1743.58276367,    -3.4284749 ,  1132.13330078, -1558.73864746,   861.73510742,  -949.86090088, -1874.14453125]], shape=(1, 10), dtype=float32, device='native:0')
Mismatch: 10 / 10 (100%)
Check `ok_cnt' == `test_case.outputs.size()' failed! in VerifyOutputs at ../tools/run_onnx.cc:499: (0 vs 1)
Aborted (core dumped)
```

**Comment by take-cheeze** [comment]
Seems generated weight is wrong

**Comment by take-cheeze** [comment]
In per channel quantization some `Mul` (and `Add` ?) op seems to have wrong shape so need to reshape to (C) to (C, 1, 1) for broadcasting.

**Pull request merged**